src/dappsExplorer/search/FavoriteDappsSection.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import { StyleSheet, Text, View } from 'react-native'
+import { useSelector } from 'react-redux'
+import { favoriteDappsWithCategoryNamesSelector } from 'src/dapps/selectors'
+import { ActiveDapp, DappSection } from 'src/dapps/types'
+import DappCard from 'src/dappsExplorer/DappCard'
+import NoResults from 'src/dappsExplorer/search/NoResults'
+import { calculateSearchScore } from 'src/dappsExplorer/utils'
+import StarIllustration from 'src/icons/StarIllustration'
+import Colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+
+interface Props {
+  onPressDapp: (dapp: ActiveDapp) => void
+  searchTerm: string
+  setFavoriteResultsEmpty: (empty: boolean) => void
+}
+
+export function FavoriteDappsSection({ onPressDapp, searchTerm, setFavoriteResultsEmpty }: Props) {
+  const { t } = useTranslation()
+  const favoriteDappsWithCategoryNames = useSelector(favoriteDappsWithCategoryNamesSelector)
+
+  // Filter dapps that match the search query if present
+  const favoritedDappsSearched = favoriteDappsWithCategoryNames.filter((dapp) =>
+    searchTerm === '' ? true : calculateSearchScore(dapp, searchTerm)
+  )
+
+  // Sort these dapps by their search score
+  favoritedDappsSearched.sort(
+    (a, b) => calculateSearchScore(b, searchTerm) - calculateSearchScore(a, searchTerm)
+  )
+
+  useEffect(() => {
+    if (favoritedDappsSearched.length > 0 && searchTerm !== '') {
+      setFavoriteResultsEmpty(false)
+    } else {
+      setFavoriteResultsEmpty(true)
+    }
+  }, [favoritedDappsSearched])
```

Comment:
```suggestion
  }, [favoritedDappsSearched, searchTerm])
```

Reply:
Fixed in d3a9def639513851398c015134bd7fa503cdd10f!